src/Shared/CoreCLRAssemblyLoader.cs

```diff
@@ -63,11 +72,29 @@ public Assembly LoadFromPath(string fullPath)
             }
         }
 
+        private Assembly TryGetWellKnownAssembly(AssemblyLoadContext context, AssemblyName assemblyName)
+        {
+            if (!_wellKnownAssemblyNames.Contains(assemblyName.Name))
+            {
+                return null;
+            }
+
+            assemblyName.Version = _currentAssemblyVersion;
+
+            return AssemblyLoadContext.Default.LoadFromAssemblyName(assemblyName);
```

Comment:
@cdmihai If that call fails then something has gone seriously wrong with your app--most likely you're missing core, required MSBuild binaries. I'm not particularly worried about the behavior in that case.